### PR TITLE
update configs to run testcafe against v3

### PR DESCRIPTION
**Pull request by lesterchoi-okta**
## Description:

Update webpack and related configs to run existing TestCafe suite against widget v3.

- limit v3 engines node >=14 <16 due to fibers deprecation in 16
- load widgetrc from playground configs using relative path
- replace deprecated onBeforeSetupMiddleware webpack function

## PR Checklist

- [ ] Have you verified the basic functionality for this change?

- [ ] Added unit tests?

- [ ] Added e2e tests

- [ ] Did you follow our [Security Best Practices](https://oktawiki.atlassian.net/wiki/display/eng/Security+Best+practices)?

- Did you verify the change by running downstream monolith artifacts? (YES | NO | UNSURE)

### Issue:

- [OKTA-508218](https://oktainc.atlassian.net/browse/OKTA-508218)

### CODE REVIEW

File: src/v3/preact.config.js [hunk 1]
```diff
@@ -10,54 +10,47 @@
  * See the License for the specific language governing permissions and limitations under the License.
  */
 
-/* eslint-disable import/no-extraneous-dependencies */
-import buffer from 'buffer';
+/* eslint-disable import/no-extraneous-dependencies,no-param-reassign */
```
Comment (shawyu-okta) [inline src/v3/preact.config.js:13]: Could we still keep the eslint-disable for `no-param-reassign` at the places they occur?

File: src/v3/preact.config.js [hunk 2]
```diff
@@ -71,15 +64,15 @@ export default {
       if (rule.test && rule.test.test('.svg')) {
         rule.exclude = /\.svg$/;
       }
+
+      // FIXME remove ts-loader
```
Comment (shawyu-okta) [inline src/v3/preact.config.js:68]: Can we file a tech debt ticket for this and link it?

File: src/v3/preact.config.js [hunk 3]
```diff
@@ -128,76 +121,60 @@ export default {
...
-          '/app/UserHome',
-          '/oauth2/v1/authorize',
-          '/auth/services/',
-          '/.well-known/webfinger'
-        ],
-        target: `http://${HOST}:${MOCK_SERVER_PORT}`
-      }],
-    },
+    // https://github.com/APIDevTools/json-schema-ref-parser#browser-support
+    config.plugins.push(new DefinePlugin({ process: { browser: true }, Buffer }));
```
Comment (shawyu-okta) [inline src/v3/preact.config.js:149]: we'll be removing this in the remove jsonforms branch so you can avoid updating this if you want

**Comment by shuowu-okta** [comment]
Please use the playground as the only test harness app for testcafe, issues with current approach:
1) maintaining two testapp can make logic run out of sync pretty easily.
2) testcafe uses different approach to mock response (mock at test stage), which means the mocks that use in test is a different set of mocks from v3 test app. One from /playground, one from /v3/src/mocks. So when issues happen in test, it would be pretty difficult for devs to debug the issue locally in the test harness app, as it cannot easily serve mocks from /playground

As we discussed before, the approach should be:
* build development widget bundle of v3
* server it from the playground harness app
* run testcafe against the playground app that loads the v3 bundle

File: webpack.playground.config.js [hunk 4]
```diff
@@ -102,28 +102,14 @@ module.exports = {
...
-      );
-      const watch = [
-        path.resolve(__dirname, 'playground', 'mocks')
-      ];
-      const env = {
-        MOCK_SERVER_PORT,
-        DEV_SERVER_PORT
-      };
+    // https://webpack.js.org/configuration/dev-server/#devserversetupmiddlewares
+    setupMiddlewares(middlewares) {
```
[the remark this answers is not in the document]
Comment (lesterchoi-okta) [reply webpack.playground.config.js:106]: deprecation notice. wanted to reduce noise while i was in the file

File: src/v3/preact.config.js (hunk 1 above)
Comment (lesterchoi-okta) [reply src/v3/preact.config.js:13]: Actually this file's primary purpose is to modify the `config` object so it's throughout

File: src/v3/preact.config.js (hunk 3 above)
Comment (lesterchoi-okta) [reply src/v3/preact.config.js:149]: Saw that. I'll deal with it at merge time

File: src/v3/preact.config.js [hunk 5]
```diff
@@ -128,76 +121,60 @@ export default {
...
+      devServer: {
+        headers: {
+          'Content-Security-Policy': mergeContentSecurityPolicies(
+            "object-src 'self'; script-src 'self' 'unsafe-eval'",
+            `script-src ${config.devServer.host}:${config.devServer.port}`,
+            playgroundConfig.devServer.headers['Content-Security-Policy'] || '',
+          ),
+        },
+      },
+    }));
```
Comment (shawyu-okta) [inline src/v3/preact.config.js:178]: There's a bunch of nesting here that makes it hard to parse what is happening, could it be broken up into smaller statements?

**Review by shawyu-okta — APPROVED** [review]
